### ISSUE TITLE
Support gateway peer load balancing in the new fabric peer gateway connector

### ISSUE THREAD

**Issue by davidkel** [hyperledger/caliper#1210]
### Please share the technical limitation of Caliper that you encountered.

https://github.com/hyperledger/caliper/issues/1205 implements a base connector but one of the points of interest will be how can performance be improved from a client application if you can load balance the calls to multiple gateways.

### Please detail your feature idea that could alleviate the limitation.

The connector could be told about multiple peers that can act as gateway peers and perform a round robin load balancing action to determine the benefit of adding more gateways peers to an organisation's network.

### Please share some details about your use case if possible, and how the new feature would make Caliper a better performance benchmarking framework.

This will provide a built in mechanism into caliper that would allow a user to experiment with the number of gateway peers to see the effect and help to plan a network.

### Please share any suggestions about the new feature's code/configuration API (using formatted YAML segments or pseudo-code).

The format was already described in https://github.com/hyperledger/caliper/issues/1205 where more than 1 peer can be added to a network config file. This feature request will exploit that whereas the base connector would only use the first peer. 

It should also be possible to do the same if multiple peers are defined in a connection profile whereas the base implementation would only use the first one it found.

**Comment by davidkel** [comment]
A roundrobin mechanism is the simplest, but it may be good to also support weighting as well so we can control load distribution

**Comment by davidkel** [comment]
this may be of interest
https://github.com/grpc/grpc/blob/master/doc/load-balancing.md
https://grpc.io/blog/grpc-load-balancing/

**Comment by leonardotkimura** [comment]
I agree! I am investigating the effect of new peers in the same organization, and this feature would be useful!

**Comment by davidkel** [comment]
see #1692 as this would allow a workload to implement it's own load balancing logic. Note however that the load balancing in both cases would be at a worker level. In reality fabric networks would almost certainly implement load balancing closer to the peers not at the client side so although a nice idea to have some sort of inbuilt load balancing for research purposes it's unlikely clients would do it in the real world. I think #1692 would be sufficient for Caliper and then this issue could be closed.

**Comment by davidkel** [comment]
Looking at the grpc option for load balancing, it relies on the DNS resolver to return multiple IPs (with ports) for a single name. This is a more complex scenario than having 2 peers with externally different names which is a more common scenario compared to an environment where internally the peers are joined with 2 different names to a channel but are presented as a single name resolving to 2 ip addresses (not sure how easy this would be to setup either if at all possible). I think the option of using grpc isn't worth persuing at this time.
